### ISSUE TITLE
CNN on 6x8 data in R

### ISSUE THREAD

**Issue by mg64ve** [dmlc/mxnet#3607]
Dear team,

I am trying to use a CNN on a dataset with 6x8 data. I am working in R.
The dataset has the following header:

`feat1_01, feat1_02, feat1_03, feat1_04, feat1_05, feat1_06, feat1_07, feat1_08, feat2_01, feat2_02, feat2_03, feat2_04, feat2_05, feat2_06, feat2_07, feat2_08, feat3_01, feat3_02, feat3_03, feat3_04, feat3_05, feat3_06, feat3_07, feat3_08, feat4_01, feat4_02, feat4_03, feat4_04, feat4_05, feat4_06, feat4_07, feat4_08, feat5_01, feat5_02, feat5_03, feat5_04, feat5_05, feat5_06, feat5_07, feat5_08, feat6_01, feat6_02, feat6_03, feat6_04, feat6_05, feat6_06, feat6_07, feat6_08, objective
 -0.74629, 0.17357, -0.03206, -0.02836, -0.01762, 0.26987, 0.99520, -0.04293, 1.00000, 0.97841, 0.68888, 0.79276, 0.48054, 0.67120, 0.55006, 0.55094, -0.18272, -0.66293, -0.04991, 0.23771, 0.69086, -0.01079, -0.50588, -0.53587, -0.51685, -0.51547, -0.28356, -0.48355, -0.23543, -0.45552, -0.31626, -0.22796, -0.63986, -0.61482, -0.40867, -0.57193, -0.31578, -0.53067, -0.41354, -0.37138, -0.99783, -0.94881, -0.77554, -0.89344, -0.69781, -0.85418, -0.75107, -0.71303, -1.0
 -0.43378, -0.74629, 0.17357, -0.03206, -0.02836, -0.01762, 0.26987, 0.99520, 0.88242, 1.00000, 0.97841, 0.68888, 0.79276, 0.48054, 0.67120, 0.55006, -0.65165, -0.18272, -0.66293, -0.04991, 0.23771, 0.69086, -0.01079, -0.50588, -0.27281, -0.51685, -0.51547, -0.28356, -0.48355, -0.23543, -0.45552, -0.31626, -0.48579, -0.63986, -0.61482, -0.40867, -0.57193, -0.31578, -0.53067, -0.41354, -0.90396, -0.99783, -0.94881, -0.77554, -0.89344, -0.69781, -0.85418, -0.75107, 1.0`

I am attaching the full dataset.csv [dataset.zip](https://github.com/dmlc/mxnet/files/548954/dataset.zip)
the code I am using is the following:

`library(mxnet)
# Data preparation
d <- read.csv('dataset.csv', sep=",", header=T)
train<-d[100:1000,]
test<-d[1001:1100,]

train<-data.matrix(train)
test<-data.matrix(test)
train.x<-train[,-ncol(train)]
train.y<-train[,ncol(train)]
train.x<-t(train.x)
test_org<-test
test<-test[,-ncol(test)]
test<-t(test)
#table(train.y)

# Convolutional NN
data <- mx.symbol.Variable('data')
# first conv
conv1 <- mx.symbol.Convolution(data=data, kernel=c(4,4), num_filter=20)
tanh1 <- mx.symbol.Activation(data=conv1, act_type="tanh")
pool1 <- mx.symbol.Pooling(data=tanh1, pool_type="max",
                           kernel=c(2,2), stride=c(2,2))
# second conv
conv2 <- mx.symbol.Convolution(data=pool1, kernel=c(4,4), num_filter=50)
tanh2 <- mx.symbol.Activation(data=conv2, act_type="tanh")
pool2 <- mx.symbol.Pooling(data=tanh2, pool_type="max",
                           kernel=c(2,2), stride=c(2,2))
# first fullc
flatten <- mx.symbol.Flatten(data=pool2)
fc1 <- mx.symbol.FullyConnected(data=flatten, num_hidden=50) 

tanh3 <- mx.symbol.Activation(data=fc1, act_type="tanh")
# second fullc
fc2 <- mx.symbol.FullyConnected(data=tanh3, num_hidden=10)
# loss
lenet <- mx.symbol.SoftmaxOutput(data=fc2)
train.array <- train.x
dim(train.array) <- c(6, 8, 1, ncol(train.x))
test.array <- test
dim(test.array) <- c(6, 8, 1, ncol(test))
mx.set.seed(0)
tic <- proc.time()
model <- mx.model.FeedForward.create(lenet, X=train.array, y=train.y,
                                     ctx=device.cpu, num.round=20, array.batch.size=100,
                                     learning.rate=0.05, momentum=0.9, wd=0.00001,
                                     eval.metric=mx.metric.accuracy,
                                     epoch.end.callback=mx.callback.log.train.metric(100))
`

When I run the R code I get the following:

`[22:30:31] /root/mxnet/mxnet/dmlc-core/include/dmlc/logging.h:235: [22:30:31] src/operator/./convolution-inl.h:373: Check failed: ksize_y <= dshape[2] + 2 * param_.pad[0] && ksize_x <= dshape[3] + 2 * param_.pad[1] kernel size exceed input
Error in eval(substitute(expr), envir, enclos) : 
  InferShape Error in convolution1: [22:30:31] src/operator/./convolution-inl.h:373: Check failed: ksize_y <= dshape[2] + 2 * param_.pad[0] && ksize_x <= dshape[3] + 2 * param_.pad[1] kernel size exceed input
Calls: mx.model.FeedForward.create ... mx.model.init.params -> mx.symbol.infer.shape -> <Anonymous> -> .External
`

but I do not understand what is the reason.
May I ask you if you see any discrepancy in my code?
Thanks.
Cheers.

**Comment by piiswrong** [comment]
your input is too small/kernel is too big

**Comment by thirdwing** [comment]
I think you can try to remove the second convolution.

**Comment by mg64ve** [comment]
Let me try.
I will let you know.
Thanks

**Comment by Hazem-azim** [comment]
mg64ve : What happened to your trials? did it work ? as I have the same issues when starting to work with MXNet?
